pkg/postgres/status_test.go

```diff
@@ -94,108 +90,105 @@ var _ = Describe("PostgreSQL status", func() {
+		Expect(podList.ReportingMightBeUnavailable(podList[0].Pod.Name)).To(BeFalse())
+		Expect(podList.ReportingMightBeUnavailable(podList[1].Pod.Name)).To(BeFalse())
 		Expect(podList.InstancesReportingStatus()).To(BeEquivalentTo(0))
-		podList.Items[1].MightBeUnavailable = true
-		Expect(podList.ReportingMightBeUnavailable(podList.Items[0].Pod.Name)).To(BeFalse())
-		Expect(podList.ReportingMightBeUnavailable(podList.Items[1].Pod.Name)).To(BeTrue())
+		podList[1].MightBeUnavailable = true
+		Expect(podList.ReportingMightBeUnavailable(podList[0].Pod.Name)).To(BeFalse())
+		Expect(podList.ReportingMightBeUnavailable(podList[1].Pod.Name)).To(BeTrue())
 		Expect(podList.InstancesReportingStatus()).To(BeEquivalentTo(1))
-		podList.Items[0].MightBeUnavailable = true
-		Expect(podList.ReportingMightBeUnavailable(podList.Items[0].Pod.Name)).To(BeTrue())
-		Expect(podList.ReportingMightBeUnavailable(podList.Items[1].Pod.Name)).To(BeTrue())
+		podList[0].MightBeUnavailable = true
+		Expect(podList.ReportingMightBeUnavailable(podList[0].Pod.Name)).To(BeTrue())
+		Expect(podList.ReportingMightBeUnavailable(podList[1].Pod.Name)).To(BeTrue())
 		Expect(podList.InstancesReportingStatus()).To(BeEquivalentTo(2))
 	})
 
-	Describe("when sorted", func() {
+	Context("when sorted", func() {
 		sort.Sort(&list)
 
 		It("primary servers are come first", func() {
-			Expect(list.Items[0].IsPrimary).To(BeTrue())
-			Expect(list.Items[0].Pod.Name).To(Equal("server-10"))
+			Expect(list[0].IsPrimary).To(BeTrue())
+			Expect(list[0].Pod.Name).To(Equal("server-10"))
 		})
 
 		It("contains the more updated server as the second element", func() {
-			Expect(list.Items[1].IsPrimary).To(BeFalse())
-			Expect(list.Items[1].Pod.Name).To(Equal("server-40"))
+			Expect(list[1].IsPrimary).To(BeFalse())
+			Expect(list[1].Pod.Name).To(Equal("server-40"))
 		})
 
 		It("contains other servers considering their status", func() {
-			Expect(list.Items[2].Pod.Name).To(Equal("server-30"))
-			Expect(list.Items[3].Pod.Name).To(Equal("server-20"))
+			Expect(list[2].Pod.Name).To(Equal("server-30"))
+			Expect(list[3].Pod.Name).To(Equal("server-20"))
 		})
 
 		It("put the non-ready servers after the ready ones", func() {
-			Expect(list.Items[4].Pod.Name).To(Equal("server-06"))
-			Expect(list.Items[4].Pod.Name).To(Equal("server-06"))
+			Expect(list[4].Pod.Name).To(Equal("server-06"))
+			Expect(list[4].Pod.Name).To(Equal("server-06"))
 		})
 
 		It("put the incomplete statuses at the bottom of the list", func() {
-			Expect(list.Items[5].Pod.Name).To(Equal("server-04"))
-			Expect(list.Items[5].Pod.Name).To(Equal("server-04"))
+			Expect(list[5].Pod.Name).To(Equal("server-04"))
+			Expect(list[5].Pod.Name).To(Equal("server-04"))
 		})
 	})
 })
 
 var _ = Describe("PostgreSQL status real", func() {
 	f, err := os.Open("testdata/lsn_overflow.json")
+	Expect(err).ToNot(HaveOccurred())
 	defer func() {
 		_ = f.Close()
 	}()
-	Expect(err).ToNot(HaveOccurred())
+	defer GinkgoRecover()
```

Comment:
Am I missing something, or should this not be needed as we are not in a goroutine?

Reply:
Pretty surprised myself. Seems to be necessary because of `json.NewDecoder(f).Decode(&list)`
I wonder if, under the covers, that executes as a goroutine.
Anyway, plan to remove this or explain it better.